core/dbt/events/functions.py

```diff
@@ -11,17 +11,53 @@
 def fire_event(e: Event) -> None:
     EVENT_HISTORY.append(e)
     if isinstance(e, CliEventABC):
-        if e.level_tag() == 'test':
+        if e.level_tag() == 'test' and not isinstance(e, ShowException):
             # TODO after implmenting #3977 send to new test level
             logger.GLOBAL_LOGGER.debug(logger.timestamped_line(e.cli_msg()))
-        elif e.level_tag() == 'debug':
+        elif e.level_tag() == 'test' and isinstance(e, ShowException):
+            # TODO after implmenting #3977 send to new test level
+            logger.GLOBAL_LOGGER.debug(
+                logger.timestamped_line(e.cli_msg()),
+                exc_info=e.exc_info,
+                stack_info=e.stack_info,
+                extra=e.extra
+            )
+        elif e.level_tag() == 'debug' and not isinstance(e, ShowException):
             logger.GLOBAL_LOGGER.debug(logger.timestamped_line(e.cli_msg()))
-        elif e.level_tag() == 'info':
+        elif e.level_tag() == 'debug' and isinstance(e, ShowException):
+            logger.GLOBAL_LOGGER.debug(
+                logger.timestamped_line(e.cli_msg()),
+                exc_info=e.exc_info,
+                stack_info=e.stack_info,
+                extra=e.extra
+            )
+        elif e.level_tag() == 'info' and not isinstance(e, ShowException):
             logger.GLOBAL_LOGGER.info(logger.timestamped_line(e.cli_msg()))
-        elif e.level_tag() == 'warn':
+        elif e.level_tag() == 'info' and isinstance(e, ShowException):
+            logger.GLOBAL_LOGGER.info(
+                logger.timestamped_line(e.cli_msg()),
+                exc_info=e.exc_info,
+                stack_info=e.stack_info,
+                extra=e.extra
+            )
+        elif e.level_tag() == 'warn' and not isinstance(e, ShowException):
             logger.GLOBAL_LOGGER.warning()(logger.timestamped_line(e.cli_msg()))
```

Comment:
parenthesis are off after `warning` but I know you mentioned this possibly being fixed in another PR.

Reply:
still worth fixing wherever we see it. thanks!